CDCgov/prime-reportstream
# Experience/validation mvp content fixes

This PR introduces:

* updates to the form label on the upload and validate pages that
  * more specifically notes which file type and extension should be used for the upload for the current sender
  * specify which hl7 version should be used in the case of hl7
* a loading state for the page while fetching organization and sender information from the api
* an error state for the page for cases where
  * an organization is selected that has no sender, or no sender that matches the user's organization by name
  * no organization is selected


## Test Steps:

1. start servers on master with `yarn run start:localdev` from `frontend-react` and `./gradlew run` from `prime-router`
2. visit http://localhost:3000/admin/settings and log in as necessary
4. activate the `user-upload` feature flag via the `admin -> feature flags` navigation
5. refresh
3. set up your user up to spoof 'hcintegrations'
8. click `User Upload` in the nav bar under `Upload & Validate`
9. _VERIFY_: upload screen shows new verbiage, specifically targeting HL7 (see screenshot below)
10. set up your user to spoof `ak-phd`
8. click `User Upload` in the nav bar under `Upload & Validate`
12. _VERIFY_: upload screen shows error banner saying that your organization has no senders (see screenshot below)

## Screenshots

### Org with no Sender

<img width="1380" alt="Screen Shot 2022-08-01 at 5 22 25 PM" src="https://user-images.githubusercontent.com/92806979/182248954-9d3eb197-05e6-49dd-8661-0a7f256650d8.png">

## New verbiage
<img width="1341" alt="Screen Shot 2022-08-02 at 12 48 26 PM" src="https://user-images.githubusercontent.com/92806979/182429847-e0fab256-b6a9-42fc-a3d1-ce5fd899339b.png">



## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
